scripts/west_commands/run_common.py

```diff
@@ -165,21 +194,113 @@ def do_run_common(command, user_args, user_runner_args, domains=None):
+                soc_yaml_file = Path(cache['SOC_FULL_DIR']) / 'soc.yml'
+                board_yaml_file = Path(cache['BOARD_DIR']) / 'board.yml'
+                group_type = 'boards'
+
+                # Search for flash runner configuration, board takes priority over SoC
+                try:
+                    with open(board_yaml_file, 'r') as f:
+                        data_yaml = yaml.safe_load(f.read())
+
+                except FileNotFoundError:
+                    continue
+
+                if 'runners' not in data_yaml:
+                    # Check SoC file
+                    group_type = 'qualifiers'
+                    try:
+                        with open(soc_yaml_file, 'r') as f:
+                            data_yaml = yaml.safe_load(f.read())
+
+                    except FileNotFoundError:
+                        continue
+
+                processed_boards[cache['BOARD_DIR']] = True
+
+                if 'runners' not in data_yaml or 'run_once' not in data_yaml['runners']:
+                    continue
+
+                for grp in data_yaml['runners']['run_once']:
+                    for cmd, data in grp.items():
+                        for group in data['groups']:
+                            run_first = bool(data['run'] == 'first')
+                            used_cmds.append(UsedFlashCommand(cmd, group[group_type], data['runners'], run_first))
+
+            if 'SOC_FULL_DIR' in cache  and len(used_cmds) > 0:
+                # Replace regex board names with full board names here so that commands can be
+                # tracked
+                for i, entry in enumerate(used_cmds):
+                    for l, match in enumerate(entry.boards):
+                        if match[0:1] == '/' and match[:1] == '/' and \
+                           re.match(match[1:len(match)-1], board) is not None:
+                            used_cmds[i].boards[l] = board
+
+    # Remove empty entries and boards that are not in the current set of images from the list
+    i = len(used_cmds) - 1
+    while i >= 0:
+        l = len(used_cmds[i].boards) - 1
+        while l >= 0:
+            if used_cmds[i].boards[l][0:1] == '/' and used_cmds[i].boards[l][:1] == '/':
+                del used_cmds[i].boards[l]
+            elif used_cmds[i].boards[l] not in board_names:
+                del used_cmds[i].boards[l]
+
+            l = l - 1
+
+        if len(used_cmds[i].boards) == 0:
+            del used_cmds[i]
+
+        i = i - 1
 
     for d in domains:
-        do_run_common_image(command, user_args, user_runner_args, d.build_dir)
+        do_run_common_image(command, user_args, user_runner_args, d.build_dir,
+                            used_cmds, board_image_count)
+
 
-def do_run_common_image(command, user_args, user_runner_args, build_dir=None):
+def do_run_common_image(command, user_args, user_runner_args, build_dir=None,
+                        used_cmds=None, board_image_count=None):
```

Comment:
These arguments can't really be None, so they should be mandatory.

Reply:
used_cmds, board_image_count):
    ^^^^^^^^^
SyntaxError: non-default argument follows default argument

Reply:
Right, you can move them before `board_dir`